### DISCUSSION TITLE
Set variables in playground from dataset

### DISCUSSION THREAD

**Discussion by hbenyamina** [langfuse/langfuse#2738] [Ideas]
I find it a bit hard to have to replace variables all the time when using the playground. Is there a way we can use the datasets to fill in the variables easily? 
Thanks.

**Comment by marcklingen** [comment]
thanks for sharing, great idea @hbenyamina! this is not supported yet but on our roadmap for the playground

**Comment by analyst3dev** [comment]
Upvote. This will make a huge difference to truly unlock business users to be able to iterate on prompts and test prompt outputs independently. (A bit like the Anthropic workbench)

**Comment by wanderslth** [comment]
Upvote on this from a biz user. :)  Having stored snippets in playground would be very handy for prompt templates like:  `{{universal snippet 1}} + unique prompt criteria + {{universal snippet 2}} + {{sample data snippet 1}} + {{sample data snippet 2}}`

**Comment by gauthierhavet** [comment]
same here// so tedious to to that.. my team can't use it

**Reply by marcklingen** [reply]
thanks for the +1

**Comment by kzhu** [comment]
+1 Big hurdle for non-technical users. I have to do this step for them each time but they are not able to do this themselves

**Comment by aaronspring** [comment]
+1 so non tech users can use prompt playground themselves. So far had to build gradio app for live edits to prompt with real data input fetched from databases.

**Comment by shutootaki** [comment]
Like others, I would like to see this feature implemented.
I think this feature would be useful for performing simple evaluations at low cost.

**Comment by ch-au** [comment]
+1 this would be an extremely helpful feature for quick prompt tests (especially more complex prompts).

**Comment by koheikameyama** [comment]
We would also greatly benefit from this feature. Currently implementing Langfuse Prompt Management for our production batch jobs.

## Our Use Case

We manage prompts with complex variables (8-10 variables per prompt). Testing in the Playground requires manually filling in all variables each time, which is time-consuming and error-prone.

## Current Workaround

We're storing example values in the `config` field as a workaround:

```json
{
  "provider": "openai",
  "model": "gpt-5-mini",
  "temperature": 0.7,
  "max_tokens": 1500,
  "example_variables": {
    "store_name": "Tokyo Store",
    "scores_summary": "Pride: 4.5, Safety: 4.2, Connection: 4.0...",
    "daily_scores_count": "7"
  }
}
```

This serves as documentation but doesn't auto-fill the Playground.

## Impact

- **Non-technical users** struggle to test prompts independently
- **Iteration speed** is significantly reduced
- **Testing accuracy** suffers from manual copy-paste errors

Having this feature would enable our business users to confidently test and iterate on prompts without engineering support. Very much looking forward to this! 🙏
